src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -101,6 +103,65 @@ Mass Segment::Solution::computeDeltaMass() const
     return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
 }
 
+Array<MatrixXd> Segment::Solution::computeAccelerations() const
+{
+    // Each MatrixXd contains the acceleration for a single dynamics for all the segment states
+    // TBI: Return contributions in the same order as the dynamics, might want to standardize later OR use a map with
+    // dynamics name (but need to make sure they are unique and existing)
+    Array<MatrixXd> accelerations = Array<MatrixXd>::Empty();
+    Size stateSize = states.getSize();
+
+    for (const auto& singleDynamics : dynamics)
```

Comment:
Can you not make this a Dynamics SPtr type and rely on polymorphism?

Reply:
so not sure what you mean here, singleDynamics is of type `Shared<Dynamics>` already and I am using the function (common to all `Dynamics` classes) to `computeContribution`, is that what you meant?

Reply:
I think using `auto` when looping through an Array is fine for readibility. but yeah, can always be explicit.